python_modules/dagster/dagster/core/execution/plan/active.py

```diff
@@ -402,7 +402,9 @@ def handle_event(self, dagster_event: DagsterEvent) -> None:
         elif dagster_event.is_step_success:
             self.mark_success(step_key)
         elif dagster_event.is_step_skipped:
-            self.mark_skipped(step_key)
+            raise DagsterInvariantViolationError(
```

Comment:
This is aggresive, but my current understanding is that no skip events originate outside of the ActiveExecution, and so this should never be called. If that's not the case, then the StepDelegatingExecutor would have a hard to diagnose bug.

Reply:
maybe worth a comment here explaining for people who don't have the context of the diff summary